client/components/site-selector/index.jsx

```diff
@@ -480,7 +480,16 @@ const navigateToSite =
 				const path = allSitesPath.replace( /^\/posts\b(\/my)?/, postsBase );
 
 				// There is currently no "all sites" version of the insights page
-				return path.replace( /^\/stats\/insights\/?$/, '/stats/day' );
+				if ( path.match( /^\/stats\/insights\// ) ) {
```

Comment:
This regular expression never matches `/stats/insights`. The trailing slash is removed on load, so I never see `/stats/day` by clicking on `All My Sites`.

```suggestion
				if ( path.match( /^\/stats\/insights\/?/ ) ) {
```

Reply:
Thanks, I removed the slash from the regex, so it brought the current behavior back. Did you see any case in which we would need to check for the optional slash?

Reply:
I did not, I think it was done for backward-compatibility reasons... I just errored on the safe side.

Reply:
Good point, I've changed the regex.